crates/re_query/src/archetype_view.rs

```diff
@@ -108,8 +108,9 @@ impl ComponentWithInstances {
         instance_keys: impl IntoIterator<Item = impl Into<::std::borrow::Cow<'a, InstanceKey>>>,
         values: impl IntoIterator<Item = impl Into<::std::borrow::Cow<'a, C>>>,
     ) -> ComponentWithInstances {
-        let instance_keys = InstanceKey::to_arrow(instance_keys);
-        let values = C::to_arrow(values);
+        // Unwrap: If the data is valid for the native types, it's valid in serialized form.
```

Comment:
And if they are not? :/

I believe `to_arrow` never fails for our standard Rerun components, but that doesn't mean the same is true for all components, correct?

It looks like this function is only used in tests, so changing it to returning a `Result` and then unwrapping in the tests seems like a good choice, or renaming it `from_native_or_die` (and explaining when it panics in the docstring), which should hopefully scare away people from accidentally using it in our production code